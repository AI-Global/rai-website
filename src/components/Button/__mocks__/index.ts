export const dummyButton = {
  action: {
    title: "Dummy Button",
    url: "/dummy",
  },
};
