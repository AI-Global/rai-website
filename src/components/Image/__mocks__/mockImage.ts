export const mockImage = {
  contentful_id: "12345",
  gatsbyImageData: {
    images: {
      sources: [
        {
          srcSet:
            "//images.ctfassets.net/rz1q59puyoaw/7GDSSUgdGoG1l2HEfVSrxz/0b8303fcaa1f283b1acd8f38be624dfb/rai-badge-full.png?w=84&h=84&q=50&fm=webp 84w,\n//images.ctfassets.net/rz1q59puyoaw/7GDSSUgdGoG1l2HEfVSrxz/0b8303fcaa1f283b1acd8f38be624dfb/rai-badge-full.png?w=169&h=170&q=50&fm=webp 169w,\n//images.ctfassets.net/rz1q59puyoaw/7GDSSUgdGoG1l2HEfVSrxz/0b8303fcaa1f283b1acd8f38be624dfb/rai-badge-full.png?w=337&h=338&q=50&fm=webp 337w",
          sizes: "(min-width: 337px) 337px, 100vw",
          type: "image/webp",
        },
      ],
      fallback: {
        src:
          "//images.ctfassets.net/rz1q59puyoaw/7GDSSUgdGoG1l2HEfVSrxz/0b8303fcaa1f283b1acd8f38be624dfb/rai-badge-full.png?w=337&h=338&q=50&fm=png",
        srcSet:
          "//images.ctfassets.net/rz1q59puyoaw/7GDSSUgdGoG1l2HEfVSrxz/0b8303fcaa1f283b1acd8f38be624dfb/rai-badge-full.png?w=84&h=84&q=50&fm=png 84w,\n//images.ctfassets.net/rz1q59puyoaw/7GDSSUgdGoG1l2HEfVSrxz/0b8303fcaa1f283b1acd8f38be624dfb/rai-badge-full.png?w=169&h=170&q=50&fm=png 169w,\n//images.ctfassets.net/rz1q59puyoaw/7GDSSUgdGoG1l2HEfVSrxz/0b8303fcaa1f283b1acd8f38be624dfb/rai-badge-full.png?w=337&h=338&q=50&fm=png 337w",
        sizes: "(min-width: 337px) 337px, 100vw",
      },
    },
    layout: "constrained",
    width: 337,
    height: 338,
    placeholder: {
      fallback:
        "data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAMAAAC6V+0/AAACN1BMVEUAAAAuLjUuLjQwMDAvLTQvLTQvLTQvLTQvLTQvLTQvLTQvLTQvLTQvLTQvLTQvLDQvLTQvLTQvLTQvLTRFQ0ovLTQvLTQvLjQvLTQvLTQ8OkE3NTwvLTRIRkxEQkgvLTQvLTRAPkU/PUMvLTRCQEYwLjUvLTQvLTQvLTQwLjVGREovLTQvLTQzMTgvLTQvLTQvLTQvLTQvLTMvLTQ6OD8vLTRFQ0k1MzovLTQvLTQvLTRHRUtJR00vLTQvLTQvLTQvLTQvLTQvLTQvLDMvLTQvLTQvLTQvLTQvLTQvLTQvLTMvLTQvLTQvLTQvLDUvLDQvLDQvLTQvLTMvLTREQkhBP0YvLTQvLTQvLTQvLTQ4Nj1EQkkvLTQzMTgvLTQvLTQwLjUwLjQvLTQvLTRLSU8vLTQvLTQ+PUMvLTQvLTROTFI2NTswLjUvLTQvLTQvLTRPTlQvLTQvLTQvLTQvLTQvLjQvLTQvLTQvLTRBP0ZjYmcvLTQvLTQvLTQvLTQvLTQvLTQvLTQvLTMvLTQvLTQvLTMvLDQ+PUMvLTRbWV5APkRKSE5WVVpaWV47OUBgXmRUUlg2NDtMSlBvbXJfXWNwbnNWVFpVU1lcWmBRT1VhYGVQTlRvbnNZWF1NTFJYVlxYV1xXVVtTUlhJR05FQ0lBP0VzcndVVFlLSU9jYmdeXWJeXGJOTFJiYGVgXmNSUFZMS1FaWF5jYWY1MzqJh4uGhYl5eH2CgYVAPkVOTVN5eHxbWV9XVluB/3tAAAAAh3RSTlMAAAAACFdof5lQgX18m5wDcICFsvv7cQN+jvj4Uf7mj2Hn/gn4lQoeLpb4glizlfIS5A2Us2f76AzgmOf7kXK06wtkBtpOOa1r3QxjmmUBDwHcC3v66w4CKwPr+lSuoMGfr1UG9qIQ9kqK/e+eb57v/otL9GwBaoSp9vZ2RoN0nUc/BE0EAwLufufhAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAB3RJTUUH5QcBEzMuthzuMgAAAXtJREFUGBklwUtIVGEYBuD3/f7/XGaOOIIOGJogLnTjZiIIjMIbSW4sUFyIm/ACrgZBV7W3UqJFi6CF6EJBkMEMkXQxYisLHKnFoJKiqHiBQdQjOOd3Rp+HyNPMgcm5QQ4B0GUejEFgrgOAQIR5gyQ/BMYv4BEIRoWM895IYIKTgCi/cPUQ35MwElejV763p6qOrciQ+mTR3sqUrj16lrooKVRlmYq4+pI+QfGDw4fWeqxuezMqIRNS3/zskxrvEMD5pLhIS0r6lZ2G7Vs19UBaVI94mtRnFhpZQrKUKFRkncRgoh48LxxyHduGZAQ/Zd3Y/g7+yj3sWGKK+LiC7TMkkcesbkv4c1L9PWQ63DXnTjb8WtnLXdoJzzsvXzn8CqDXbuWPxDWJ/ulOtnCBnOhm08E/Lr4YI9Cy0h2ORchLj6d/Lme7PkKA2oFxvbGqdcRKbujkm9+AAn6tDn9+qhI3u7NlMtqn97egkJN8q+cDL/t/quH5u+bKJdwCMKl5a4qUi5YAAAAASUVORK5CYII=",
    },
  },
  description: "Rai Badge in Full - Description",
  title: "Rai Badge Full",
};
