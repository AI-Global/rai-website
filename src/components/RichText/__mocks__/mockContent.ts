import {
  ContentfulRichTextGatsbyReference,
  RenderRichTextData,
} from "gatsby-source-contentful/rich-text";

const mockHeadingOne = `
{
  "data":{},
  "content":[
    {
      "data":{},
      "marks":[],
      "value":"Heading 1",
      "nodeType":"text"
    }
  ],
  "nodeType":"heading-1"
}`;

const mockHeadingTwo = `
{
  "data":{},
  "content":[
    {
      "data":{},
      "marks":[],
      "value":"Heading 2",
      "nodeType":"text"
    }
  ],
  "nodeType":"heading-2"
}`;

const mockHeadingThree = `
{
  "data":{},
  "content":[
    {
      "data":{},
      "marks":[],
      "value":"Heading 3",
      "nodeType":"text"
    }
  ],
  "nodeType":"heading-3"
}`;

const mockParagraph = `
{
  "data":{},
  "content":[
    {
      "data":{},
      "marks":[],
      "value":"This is sample paragraph ",
      "nodeType":"text"
    }
  ],
  "nodeType":"paragraph"
}`;

export const mockParagraphWithLink = `
{
  "data":{},
  "content":[
    {
      "data":{},
      "marks":[],
      "value":"This is sample paragraph ",
      "nodeType":"text"
    },
    {
      "nodeType":"hyperlink",
      "content":[
        {
          "nodeType":"text",
          "value":"with a link",
          "marks":[],
          "data":{}
        }
      ],
      "data":{
        "uri":"/foo"
      }
    }
  ],
  "nodeType":"paragraph"
}`;

export const mockContent = {
  raw: `{
      "data":{},
      "content":[
        ${mockHeadingOne},
        ${mockHeadingTwo},
        ${mockHeadingThree},
        ${mockParagraph},
        ${mockParagraphWithLink}
      ],
      "nodeType":"document"
    }`,
} as RenderRichTextData<ContentfulRichTextGatsbyReference>;
