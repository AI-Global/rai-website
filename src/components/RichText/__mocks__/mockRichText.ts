import {
  ContentfulRichTextGatsbyReference,
  RenderRichTextData,
} from "gatsby-source-contentful/rich-text";

const mockHeadingOne = `
{
  "data":{},
  "content":[
    {
      "data":{},
      "marks":[],
      "value":"Heading 1",
      "nodeType":"text"
    }
  ],
  "nodeType":"heading-1"
}`;

const mockHeadingTwo = `
{
  "data":{},
  "content":[
    {
      "data":{},
      "marks":[],
      "value":"Heading 2",
      "nodeType":"text"
    }
  ],
  "nodeType":"heading-2"
}`;

const mockHeadingThree = `
{
  "data":{},
  "content":[
    {
      "data":{},
      "marks":[],
      "value":"Heading 3",
      "nodeType":"text"
    }
  ],
  "nodeType":"heading-3"
}`;

export const mockParagraph = `
{
  "data":{},
  "content":[
    {
      "data":{},
      "marks":[],
      "value":"A robot may not injure a human being or, through inaction, allow a human being to come to harm ",
      "nodeType":"text"
    },
    {
      "data":{},
      "marks":[{"type":"bold"}],
      "value":"To boldly go where no robot has gone before. ",
      "nodeType":"text"
    },
    {
      "data":{},
      "marks":[{"type":"underline"}],
      "value":"The underlying problem with robots is that they are robots. ",
      "nodeType":"text"
    },
    {
      "data":{},
      "marks":[{"type":"italic"}],
      "value":"The Robot R1 is an italian humanoid. ",
      "nodeType":"text"
    }
  ],
  "nodeType":"paragraph"
}`;

export const mockParagraphWithLink = `
{
  "data":{},
  "content":[
    {
      "data":{},
      "marks":[],
      "value":"This is sample paragraph ",
      "nodeType":"text"
    },
    {
      "nodeType":"hyperlink",
      "content":[
        {
          "nodeType":"text",
          "value":"with a link",
          "marks":[],
          "data":{}
        }
      ],
      "data":{
        "uri":"/foo"
      }
    }
  ],
  "nodeType":"paragraph"
}`;

export const mockStrongParagraph = `
{
  "data":{},
  "content":[
    {
      "data":{},
      "marks":[{"type":"bold"}],
      "value":"Dummy strong content ",
      "nodeType":"text"
    }
  ],
  "nodeType":"paragraph"
}`;

export const mockContent = {
  raw: `{
      "data":{},
      "content":[
        ${mockHeadingOne},
        ${mockHeadingTwo},
        ${mockHeadingThree},
        ${mockParagraph},
        ${mockParagraphWithLink}
      ],
      "nodeType":"document"
    }`,
} as RenderRichTextData<ContentfulRichTextGatsbyReference>;

export const dummyContent = {
  raw: `{
        "data":{},
        "content":[
          ${mockParagraph}
        ],
        "nodeType":"document"
      }`,
};

// {
//   "data":{},
//   "content":[
//     {
//       "data":{},
//     "content":[
//       {
//         "data":{},
//         "marks":[{"type":"bold"}],
//         "value":"This text should be bold",
//         "nodeType":"text"
//       },
//       {
//         "data":{},
//         "marks":[],
//         "value":", ",
//         "nodeType":"text"
//       },
//       {
//         "data":{},
//         "marks":[{"type":"italic"}],
//         "value":"This text should be italic",
//         "nodeType":"text"
//       },
//       {
//         "data":{},
//         "marks":[],
//         "value":", ",
//         "nodeType":"text"
//       },
//       {
//         "data":{},
//         "marks":[{"type":"underline"}],
//         "value":"This text should be underlined",
//         "nodeType":"text"
//       },
//       {
//         "data":{},
//         "marks":[],
//         "value":", sed do errus mod tempor in conduit ut laboratory et deplore electromagna aliqua. Ut enim ad minimum veniam, quis no indestruct exoform ullamco laboris nisi ut alius equip ex ea commando evaluant. Duis ex machina aute ire dolorus in scan detectus an voluptate volt esse cesium dolore eu futile nulla parameter. Execute primus sint occaecat cupidatat non proident, sunt in culpa qui technia deserunt mondus anim id est proceus.",
//         "nodeType":"text"
//       }
//     ],
//     "nodeType":"paragraph"
//   }
// ],
// }
