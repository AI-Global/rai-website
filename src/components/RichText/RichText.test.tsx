import React from "react";
import { cleanup, render } from "@testing-library/react";
import { RichText } from ".";
import { mockContent } from "./__mocks__/mockRichText";

afterEach(cleanup);

// TODO: Cleanup these tests. This is just to get CI to pass right now
it("renders some stuff", () => {
  const { container } = render(<RichText content={mockContent} />);
  expect(container).toMatchSnapshot();
});
