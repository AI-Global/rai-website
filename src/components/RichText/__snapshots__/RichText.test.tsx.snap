// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders some stuff 1`] = `
<div>
  <h1
    class="heading h1"
  >
    Heading 1
  </h1>
  <h2
    class="heading h2"
  >
    Heading 2
  </h2>
  <h3
    class="heading h3"
  >
    Heading 3
  </h3>
  <p
    class="text base"
  >
    A robot may not injure a human being or, through inaction, allow a human being to come to harm 
    <strong
      class="strong"
    >
      To boldly go where no robot has gone before. 
    </strong>
    <u
      class="u"
    >
      The underlying problem with robots is that they are robots. 
    </u>
    <em
      class="em"
    >
      The Robot R1 is an italian humanoid. 
    </em>
  </p>
  <p
    class="text base"
  >
    This is sample paragraph 
    <a
      class="link"
      href="/foo"
      rel="noreferrer"
      target="_blank"
    >
      with a link
    </a>
  </p>
</div>
`;
